tiledb/sm/array_schema/dimension.cc

```diff
@@ -1613,25 +1619,29 @@ Status Dimension::check_tile_extent() const {
   if (!is_int) {
     // Check if tile extent is negative or 0
     if (*tile_extent <= 0)
-      return LOG_STATUS(Status_DimensionError(
-          "Tile extent check failed; Tile extent must be greater than 0"));
+      throw DimensionException(
+          name() + " [check_tile_extent] Tile extent must be greater than 0");
 
     if (*tile_extent > (domain[1] - domain[0] + 1))
-      return LOG_STATUS(
-          Status_DimensionError("Tile extent check failed; Tile extent "
-                                "exceeds dimension domain range"));
+      throw DimensionException(
+          name() + " [check_tile_extent] Tile extent " +
+          std::to_string(*tile_extent) +
+          " exceeds range on dimension domain [" + std::to_string(domain[0]) +
+          ", " + std::to_string(domain[1]) + "]");
   } else {
     // Check if tile extent is 0
     if (*tile_extent == 0)
-      return LOG_STATUS(Status_DimensionError(
-          "Tile extent check failed; Tile extent must not be 0"));
+      throw DimensionException(
+          name() + " [check_tile_extent] Tile extent must not be 0");
```

Comment:
Same comment as above (drop `[check_tile_name]` and make it more explicit what `name()` refers to).